src/Metadata/AssemblyMetadata.Methods.cs

```diff
@@ -92,9 +92,19 @@ public BlobHandle GetMethodOrConstructorSignature(MethodBase methodBase)
                     {
                         foreach (var par in parameters)
                         {
+                            var p = parEnc.AddParameter();
                             if (par.ParameterType.IsByRef)
                             {
-                                parEnc.AddParameter().Type(true).FromSystemType(par.ParameterType.GetElementType(), this);
+                                var rts = par.GetRequiredCustomModifiers();
+                                if (rts.Length > 0)
+                                {
+                                    var pcm = p.CustomModifiers();
+                                    foreach (var rt in rts)
+                                    {
+                                        pcm = pcm.AddModifier(ResolveTypeReference(rt), false);
```

Comment:
I don't understand this line. `pcm` is never used in the following. Why do we assign? What is going on here?

Reply:
Since `AddModifier` returned instance of itself and it is a struct so I thought it might not work, I am not familiar with this code so I added as a safety.

Reply:
[CustomModifiersEncoder](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.metadata.ecma335.custommodifiersencoder.addmodifier?view=net-5.0) follows an immutable design. Thus, it could make sense to replace `pcm` by the newer version. However, in this case, you would need to consume `pcm` later in the code.